### PR TITLE
Add "empty username" check in the test

### PR DESCRIPTION
**Pull request by mourya007**
Check empty username in the  basicauth_test.go.

### CODE REVIEW

File: staging/src/k8s.io/apiserver/plugin/pkg/authenticator/request/basicauth/basicauth_test.go [hunk 1]
```diff
@@ -63,6 +63,12 @@ func TestBasicAuth(t *testing.T) {
 			ExpectedPassword: "",
 			ExpectedErr:      true,
 		},
+		"empty username": {
+			ExpectedCalled:   true,
+			ExpectedUsername: "",
+			ExpectedPassword: "mypassword",
```
Comment (roycaihw) [inline staging/src/k8s.io/apiserver/plugin/pkg/authenticator/request/basicauth/basicauth_test.go]: any testcase with non-nil `ExpectedUsername`/`ExpectedPassword` will trigger an [errInvalidAuth](https://github.com/kubernetes/kubernetes/blob/1719a9efc393dd48f1414d4949fb72000266a76e/staging/src/k8s.io/apiserver/plugin/pkg/authenticator/request/basicauth/basicauth.go#L49), as long as [Password](https://github.com/kubernetes/kubernetes/blob/1719a9efc393dd48f1414d4949fb72000266a76e/staging/src/k8s.io/apiserver/plugin/pkg/authenticator/request/basicauth/basicauth_test.go#L49) is unassigned

this file is for testing if different [Password interfaces](https://github.com/kubernetes/kubernetes/blob/1719a9efc393dd48f1414d4949fb72000266a76e/staging/src/k8s.io/apiserver/pkg/authentication/authenticator/interfaces.go#L42)' responses get handled by [Authenticator](https://github.com/kubernetes/kubernetes/blob/1719a9efc393dd48f1414d4949fb72000266a76e/staging/src/k8s.io/apiserver/plugin/pkg/authenticator/request/basicauth/basicauth.go#L39) properly. adding a case that triggers same response from an existing `Password` doesn't add much value IMO

Comment (mourya007) [reply staging/src/k8s.io/apiserver/plugin/pkg/authenticator/request/basicauth/basicauth_test.go]: Thank for the review, but what if there is a case when username is empty and password will be anything let's say "mypass:withcolon" ? Couldn't we should handle this case too ?

**Comment by mourya007** [comment]
I signed it

**Pull request closed without merging**